apps/ramp-network/src/components/RampWidget.tsx

```diff
@@ -0,0 +1,43 @@
+import { useRef, useEffect } from 'react';
+import { RampInstantSDK, RampInstantEvent } from '@ramp-network/ramp-instant-sdk';
+
+type Props = {
+  url?: string;
+  assets: string;
+  address: string;
+  onClose?: () => void;
+};
+
+const WIDGET_CLOSE_EVENT = 'WIDGET_CLOSE';
+const PURCHASE_CREATED_EVENT = 'PURCHASE_CREATED';
+
+const RampWidget = ({ url, assets, address, onClose }: Props) => {
+  const containerNode = useRef(null);
+
+  useEffect(() => {
+    new RampInstantSDK({
+      url,
+      hostAppName: 'Ramp Network Safe App',
+      hostLogoUrl: 'https://docs.ramp.network/img/logo-1.svg',
+      swapAsset: assets,
+      userAddress: address,
+      containerNode: containerNode.current || undefined,
+    })
+      .on('*', (event: RampInstantEvent) => {
+        if (event.type === WIDGET_CLOSE_EVENT) {
+          onClose?.();
+        }
+
+        if (event.type === PURCHASE_CREATED_EVENT) {
+          // TODO: Send Analytics when the infra is ready
+          // https://github.com/gnosis/safe-apps-sdk/issues/255
+          console.log('PURCHASE_CREATED_EVENT', event);
+        }
+      })
+      .show();
+  }, [address, assets, onClose, url]);
+
+  return <div id="ramp-widget-container"></div>;
```

Comment:
I'd move it to `index.html` because it always should be there in DOM, wdyt?

Reply:
mmm, I rewrote this because you are right. I can use a html component as the widget container but the way i was initialising it generates all the necessary HTML. So remove the components and created a initializeRampWidget function